### PR TITLE
BLD: circleCI- merge before build, add -n to sphinx

### PR DESCRIPTION
**Pull request by mattip**
Add a merge to the checkout step as per [this comment](https://discuss.circleci.com/t/show-test-results-for-prospective-merge-of-a-github-pr/1662/14) to merge master into the PR.

Also add a step to show how many warnings are still being emitted with `sphinx-build -n ...`, xref issue gh-13114. It should not cause the build to fail, but those who are interested can dive into the log files and see the progress.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,11 +50,23 @@ jobs:
...
+      - run:
+          name: build devdocs w/ref warnings
+          command: |
+            set +e
+            # allow this to fail for now: issue 13114
+            . venv/bin/activate
+            cd doc
+            SPHINXOPTS="-q -n" make -e html
+            # clear the error
+            echo ok
```
Comment (eric-wieser) [inline .circleci/config.yml:62]:
```suggestion
            . venv/bin/activate
            cd doc
            SPHINXOPTS="-q -n" make -e html || echo "ignoring errors for now, see gh-13114"
```

**Comment by mattip** [comment]
The added job [works](https://app.circleci.com/pipelines/github/numpy/numpy/4075/workflows/5d320a31-44aa-4c23-bcf8-bd1d2e5973e9/jobs/16165/parallel-runs/0/steps/0-106), but I am not sure the merge-with-head `post` step is working in the [checkout step](https://app.circleci.com/pipelines/github/numpy/numpy/4075/workflows/5d320a31-44aa-4c23-bcf8-bd1d2e5973e9/jobs/16165/parallel-runs/0/steps/0-101)

File: .circleci/config.yml [hunk 2]
```diff
@@ -53,9 +52,13 @@ jobs:
...
           name: build devdocs w/ref warnings
           command: |
+            set +e
+            # allow this to fail for now: issue 13114
             . venv/bin/activate
             cd doc
-            SPHINXOPTS="-q -n" make -e html || echo "ignoring errors for now, see gh-13114"
+            SPHINXOPTS="-q -n" make -e html
+            # clear the error
+            echo ok
```
Comment (mattip) [inline .circleci/config.yml:61]: see if this reports the number of warnings at the end, it was truncated before

**Comment by eric-wieser** [comment]
The downside of this PR is that it increases the doc build time. Does it make sense to do the nitpick checks _after_ building and uploading the doc artifacts, to make this extra delay only matter to those fixing warnings? Or should we leave it as is to fix all of the warnings?

**Comment by mattip** [comment]
> The downside of this PR is that it increases the doc build time.

I view this as a "temporary" step till we get the warning count down (but that could take a while). The CircleCI build time is 26 minutes (single process). Travis [is running](https://travis-ci.org/github/numpy/numpy/builds/733555259) for ~55 minutes wall clock, 2hrs50 min process time, so I think the extra ~5 minutes are not going to slow down the PR turn-around time.

**Comment by eric-wieser** [comment]
> so I think the extra ~5 minutes are not going to slow down the PR turn-around time.

Note that for doc-only PRs, the travis build time doesn't really matter. Still, 5 minutes over 25 is not a disaster.

**Review by eric-wieser — APPROVED** [review]
Let's put this in and see if we can knock down the number of warnings

**Comment by charris** [comment]
Thanks Matti. The big drain on the testing seems to be PRs with a dozen small updates.

**Pull request merged**